src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPullerService.cs

```diff
@@ -279,7 +212,19 @@ private ValueTask WaitForAvailableHandler(CancellationToken cancellationToken)
             }
         }
 
-        private static IEnumerable<string> ReadCurrentContent(Channel<string> channel)
+        private int GetReleaseDelay(int deliveryCount)
+        {
+            if (this._eventGridTopicSubscription.RetryDelays == null)
+            {
+                return (int)Math.Min(Math.Pow(2, deliveryCount - 1), int.MaxValue);
+            }
+
+            return deliveryCount - 1 < this._eventGridTopicSubscription.RetryDelays.Count
+                ? this._eventGridTopicSubscription.RetryDelays[deliveryCount - 1]
+                : this._eventGridTopicSubscription.RetryDelays.Last();
```

Comment:
Do we know if these retry options apply to push and pull model? https://learn.microsoft.com/en-us/azure/event-grid/delivery-and-retry#retry-policy

Reply:
Wondering cause otherwise it seems we could be releasing some events forever

Reply:
When configuring a subscription on a topic, you can set the maximum attempts (currently between 1-10). The lock duration is also configurable at the same place (you get 1 minute by default, and can have up to 5).

Since these value limits can change over time, I suggest to keep the current "dynamic" logic that will stretch to any retry amount configured.

![image](https://github.com/user-attachments/assets/e62c71b0-b260-4e18-b7bf-ef8de418b347)